### PR TITLE
Fix menu items for toc

### LINKED ISSUE #11553: Rename the table of content plugin id
This is a small issue to track renaming:

https://github.com/jupyterlab/jupyterlab/blob/1bf5fdff171f8a9e91755821efd376d5742778ba/packages/toc-extension/src/index.ts#L235

To:

```ts
'@jupyterlab/toc-extension:registry'
```

So it's consistent with the rest of the code base.

### Additional context

This came as an observation when looking at https://github.com/jupyterlab/jupyterlab/pull/11530

Adding to 4.0 as a potential breaking change, since the rename can affect users who were using it for disabling the extension for instance.

### PR DESCRIPTION
**Pull request by fcollonval**
## References

Fixes #11553

This corrects the plugin name for the toc extension. That fixes the missing menu items appearance (in particular the context menu).

It fixes also the default rank for context menu items (that could appears after the default `Right click` notice).

## Code changes

Mainly change @jupyterlab/toc-extension plugin id

## User-facing changes

Menus items for toc are displayed
Context menu items do not appear after the right click notice

## Backwards-incompatible changes

`@jupyterlab/toc:plugin` renamed in `@jupyterlab/toc-extension:plugin`

### CODE REVIEW

File: docs/source/extension/extension_migration.rst [hunk 1]
```diff
@@ -23,6 +23,8 @@ bumped their major version (following semver convention):
    from Blueprint JS library. Extensions using ``Button``, ``Collapse`` or ``InputGroup`` may
    need to switch to the Blueprint components as the interfaces of those components in JupyterLab
    do not match those of Blueprint JS.
+- ``@jupyterlab/toc:plugin`` renamed ``@jupyterlab/toc-extension:plugin``
```
Comment (davidbrochart) [inline docs/source/extension/extension_migration.rst:26]:
```suggestion
- ``@jupyterlab/toc:plugin`` renamed ``@jupyterlab/toc-extension:registry``
```

File: packages/toc-extension/src/index.ts [hunk 2]
```diff
@@ -232,7 +232,7 @@ async function activateTOC(
  * @private
  */
 const extension: JupyterFrontEndPlugin<ITableOfContentsRegistry> = {
-  id: '@jupyterlab/toc:plugin',
+  id: '@jupyterlab/toc-extension:plugin',
```
Comment (davidbrochart) [inline packages/toc-extension/src/index.ts:235]:
```suggestion
  id: '@jupyterlab/toc-extension:registry',
```

**Review by davidbrochart — COMMENTED** [review]
In https://github.com/jupyterlab/jupyterlab/issues/11553#issue-1063938921 @jtpio suggested renaming to `'@jupyterlab/toc-extension:registry'`.

**Review by jtpio — APPROVED** [review]
Thanks!

**Comment by jtpio** [comment]
Looks like there is a small conflict to resolve in the docs as a follow-up to the merge of https://github.com/jupyterlab/jupyterlab/pull/11537

**Comment by jtpio** [comment]
Thanks!

**Comment by fcollonval** [comment]
I'll backport it to get the fix on the context menu in 3.2.x

@meeseeksdev please backport to 3.2.x

**Pull request merged**